work17/index.js

```diff
@@ -0,0 +1,191 @@
+"use strict";
+const REQUEST_URL = "https://mocki.io/v1/d4a57e5a-8c84-4fee-aa05-70a1791c0d86";
+
+// 間違っているURLの場合↓
+// const REQUEST_URL = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// 空の配列の場合↓
+// const REQUEST_URL = "https://mocki.io/v1/242b685f-a3d7-45a8-aeca-0376bd495b89";
+// 503エラーの場合↓
+// const REQUEST_URL = "https://httpstat.us/503";
+
+const wrapDiv = document.querySelector(".slider_wrap");
+
+const renderLoading = () => {
+    const div = document.createElement("div");
+    const img = document.createElement("img");
+
+    div.classList = "loading-wrap";
+    img.src = "./img/loading-circle.gif";
+
+    wrapDiv.appendChild(div).appendChild(img);
+    document.body.appendChild(div);
+}
+
+const hideLoadingImg = () => {
+    document.querySelector(".loading-wrap").remove();
+}
+
+const init = async () => {
+    renderLoading();
+    let imgJsonData;
+    try {
+        imgJsonData = await getRequestData(REQUEST_URL);
+    } catch (error) {
+        renderErrorMessage(`${error.message}`)
+    } finally {
+        hideLoadingImg();
+    }
+
+    if (!imgJsonData.data.length) {
+        renderErrorMessage("JSONデータが空です。");
+        return;
+    }
+    renderSlideImg(imgJsonData.data);
+}
+
+const getRequestData = async REQUEST_URL => {
+    return new Promise(resolve => {
+        setTimeout(() => {
+            resolve(fetchSlideImgData(REQUEST_URL));
+        }, 3000);
+    })
+}
+
+const fetchSlideImgData = async () => {
+    const parameter = {
+        headers: {
+            'Content-Type': 'application/json'
+        }
+    }
+    const response = await fetch(REQUEST_URL, parameter);
+    if (!response.ok) {
+        renderErrorMessage(`${response.status}:${response.statusText}`);
+    } else {
+        const jsonData = response.json();
+        return jsonData;
+    }
+}
+
+const renderErrorMessage = errorMessage => {
+    const p = document.createElement("p");
+
+    p.textContent = errorMessage;
+    document.body.prepend(p);
+}
+
+const renderSlideImg = imagesDataArray => {
+    const slideDiv = document.createElement("div");
+    const ul = document.createElement("ul");
+    const fragment = document.createDocumentFragment();
+
+    ul.classList = "slider_item"
+    slideDiv.classList = "slider_slide";
+
+    for (let i = 0; i < imagesDataArray.length; i++) {
+        const li = document.createElement("li");
+        const img = document.createElement("img");
+
+        img.src = imagesDataArray[i].src;
+        li.classList = "slider_items";
+        if (!i) {
+            li.classList.add("is-display");
+        }
```

Comment:
IMO
I think it is easier to understand if I write it as follows rather than writing it in logical negation.
0 is indeed implicitly false, but I also believe that writing it explicitly makes the process easier to see
```suggestion
        i === 0 && li.classList.add("is-display");
```

Reply:
It would be easier to check my code if I put a "0".
I took your point and checked it, and I thought that if statements with logical operators are easier to read than if statements.
Thank you very much.

commit : [a84cc98](https://github.com/yasu-pro/js-handson/pull/22/commits/a84cc984e9a44566ecd9a0d79a95001cb601cf8e)